core/src/bls_sigverify/bls_sigverifier.rs

```diff
@@ -162,6 +158,27 @@ impl BLSSigVerifier {
                         continue;
                     };
 
+                    // Capture votes received metrics before old messages are potentially discarded below.
+                    match self
+                        .consensus_metrics
+                        .write()
+                        .record_vote(*solana_pubkey, &vote_message.vote)
+                    {
+                        Ok(()) => (),
+                        Err(err) => {
+                            error!(
+                                "recording vote on slot {} failed with {err:?}",
+                                vote_message.vote.slot()
+                            );
+                        }
+                    }
+                    // Only need votes newer than root slot
+                    if vote_message.vote.slot() <= root_bank.slot() {
+                        self.stats.received_old.fetch_add(1, Ordering::Relaxed);
+                        packet.meta_mut().set_discard(true);
+                        continue;
+                    }
+
```

Comment:
We put this at the beginning hoping to bypass the later operations if possible, why are we moving it to here?

Reply:
I needed to move the code to look up the sender of the vote message above as I need that info to record the metrics.